.github/workflows/validate-related-images.yaml

```
@@ -0,0 +1,44 @@
+          echo ${{ github.event.number }} > pr_number.txt
+      - name: Upload the PR number as artifact
+        if: github.event_name == 'pull_request'
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+        with:
+          name: pr_number_related_images
+          path: ./pr_number.txt
+          retention-days: 1
+      - name: Check if RELATED_IMAGE references changed
+        if: github.event_name == 'pull_request'
+        id: check
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:2:26: Double quote to prevent globbing and word splitting

(shellcheck)